core/src/main/scala/scalaz/Maybe.scala

```diff
@@ -397,6 +397,17 @@ sealed abstract class MaybeInstances extends MaybeInstances0 {
         case -\/(a) => a
         case \/-(a) => a
       }
+
+      override def altly2[Z, A1, A2](a1: => Maybe[A1], a2: => Maybe[A2])(f: A1 \/ A2 => Z): Maybe[Z] = a1 match {
```

Comment:
you still need to implement `def alt` otherwise this is incomplete. Your previous `alt` was reasonable. It is reasonable to have optimised `alt` and `altly2` for `Maybe`.